examples/raytrace-parallel/index.js

```diff
@@ -0,0 +1,111 @@
+const button = document.getElementById('render');
+const canvas = document.getElementById('canvas');
+const scene = document.getElementById('scene');
+const concurrency = document.getElementById('concurrency');
+const concurrencyAmt = document.getElementById('concurrency-amt');
+const timing = document.getElementById('timing');
+const timingVal = document.getElementById('timing-val');
+const ctx = canvas.getContext('2d');
+
+button.disabled = true;
+concurrency.disabled = true;
+
+// First up, load our wasm
+wasm_bindgen('./raytrace_parallel_bg.wasm')
+  .then(run)
+  .catch(e => {
+    if (e instanceof WebAssembly.CompileError) {
+      let msg = 'failed to compile wasm module';
+      msg += `\nerror: ${e.message}`;
+      msg += "\ncurrently this requires nightly Firefox";
+      alert(msg);
+    } else {
+      console.error(e);
+    }
+  });
+
+const { Scene, WorkerPool } = wasm_bindgen;
+
+function run() {
+  // The maximal concurrency of our web worker pool is `hardwareConcurrency`,
+  // so set that up here and this ideally is the only location we create web
+  // workers.
+  pool = new WorkerPool(navigator.hardwareConcurrency);
```

Comment:
Maybe it's just me but I took me some time to get that this is exported from wasm, I think the destructuring shouldn't be outside the function and not at all.

Reply:
This is nittiest of nitpicking but:

As long as there are multiple workers that are running on different cores, then this is parallelism.

Just because there are synchronization points doesn't make something not parallel.

concurrent = operations in-progress (but maybe not doing work) at the same time

parallel = doing work at the same time

For example, threading is always concurrent, but it might not be parallel if there is only a single core that is only actually working on one thread at any given moment and is time slicing between all of the threads.